Automattic/wp-calypso
# i18n-calypso: add the isRtl functionality and remove components/rtl

Currently the helpers in `components/rtl` use `lib/i18n-utils` to get the current locale's RTL flag, using the info from the [_client/languages.js_ module](https://github.com/Automattic/wp-calypso/blob/master/client/languages.js). But that's Calypso-specific and code that uses only `i18n-calypso` and `@automattic/components` doesn't have access to that.

Fortunately, the RTL flag is encoded in our translation files, too:
- Core translations include a special translated string `"ltr"`. If it translates to `"rtl"`, we have a RTL language!
- Calypso translations include a `[ '' ][ 'momentjs_locale' ][ 'textDirection' ]` property that has either the `ltr` or `rtl` value

I moved the `components/rtl` functionality to `i18n-calypso` and retrieve the RTL flag from the places mentioned above.

That makes the `i18n-calypso` dependency the only thing you need for your components to be localized.

**How to test:**
Verify that the RTL flag returned by `i18n.isRtl()` and the `withRtl` and `useRtl` wrappers is correct. One way to do that is to follow the testing steps from #36854 and test the Masterbar Publish popover.
